processor/src/main/java/org/mapstruct/ap/internal/model/source/ForgedMethod.java

```diff
@@ -120,6 +124,36 @@ public ForgedMethod(String name, Type sourceType, Type returnType, MapperConfigu
         this.mappingOptions = mappingOptions == null ? MappingOptions.empty() : mappingOptions;
         this.mappingOptions.initWithParameter( sourceParameter );
         this.forgedNameBased = forgedNameBased;
+        sourceReadAccessors = initSourceReadAccessors( getSourceParameters(), mappingOptions );
+    }
+
+    private Map<String, Accessor> initSourceReadAccessors( List<Parameter> sourceParameters,
```

Comment:
Can you extract this method to the `MappingMethodUtils`? The code here and in the `SourceMethod` is exactly the same.